### PR TITLE
Feature/251 rate damp

### LINKED ISSUE #251: rateDamp C++ module
**Describe your use case**
This module computes torque commands to zero the angular rates of the spacecraft.

**Describe alternatives solutions you've considered**
The same result can be achieved by setting the proportional gain of mrpPD to zero.

**Additional context**
n/a

### PR DESCRIPTION
**Pull request by rcalaon**
* **Tickets addressed:** bsk-932
* **Review:** By commit  
* **Merge strategy:** Merge (no squash)  

## Description
This module implements a negative feedback control law that zeroes the angular rates of the spacecraft hub.

## Verification
A unit test is provided to test the correctness of the output.

## Documentation
Documentation is provided

## Future work
The module can be enhanced adding different gains for the different components of the angular rate vector.

### CODE REVIEW

File: src/fswAlgorithms/attControl/rateDamp/_UnitTest/test_rateDamp.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
+#
+#  ISC License
+#
+#  Copyright (c) 2016, Autonomous Vehicle Systems Lab, University of Colorado at Boulder
```
Comment (patkenneally) [inline src/fswAlgorithms/attControl/rateDamp/_UnitTest/test_rateDamp.py]: Can update this copyright statement.

File: src/fswAlgorithms/attControl/rateDamp/rateDamp.h [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+#include "architecture/_GeneralModuleFiles/sys_model.h"
+#include "architecture/utilities/avsEigenSupport.h"
+#include "architecture/utilities/bskLogging.h"
+#include "architecture/messaging/messaging.h"
+#include "architecture/msgPayloadDefC/NavAttMsgPayload.h"
+#include "architecture/msgPayloadDefC/CmdForceBodyMsgPayload.h"
+#include "cMsgCInterface/CmdTorqueBodyMsg_C.h"
+
+
+/*! @brief A class to perform EMA SEP pointing */
```
Comment (patkenneally) [inline src/fswAlgorithms/attControl/rateDamp/rateDamp.h]: Let's check all comments in the module are correct and relevant

File: src/fswAlgorithms/attControl/rateDamp/_UnitTest/test_rateDamp.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+
+from Basilisk.utilities import SimulationBaseClass
+from Basilisk.utilities import unitTestSupport  # general support file with common unit test functions
+from Basilisk.fswAlgorithms import rateDamp  # import the module that is to be tested
+from Basilisk.utilities import macros
+from Basilisk.architecture import messaging
+from Basilisk.architecture import bskLogging
+
+# uncomment this line is this test is to be skipped in the global unit test run, adjust message as needed
```
Comment (patkenneally) [inline src/fswAlgorithms/attControl/rateDamp/_UnitTest/test_rateDamp.py]: Can delete these redundant comments.

**Review by patkenneally — APPROVED** [review]

File: src/fswAlgorithms/attControl/rateDamp/_UnitTest/test_rateDamp.py (hunk 1 above)
Comment (rcalaon) [reply src/fswAlgorithms/attControl/rateDamp/_UnitTest/test_rateDamp.py]: Done

File: src/fswAlgorithms/attControl/rateDamp/_UnitTest/test_rateDamp.py (hunk 3 above)
Comment (rcalaon) [reply src/fswAlgorithms/attControl/rateDamp/_UnitTest/test_rateDamp.py]: Done

File: src/fswAlgorithms/attControl/rateDamp/rateDamp.h (hunk 2 above)
Comment (rcalaon) [reply src/fswAlgorithms/attControl/rateDamp/rateDamp.h]: Sorry >.< done

**Pull request merged**